app/misc/git.ts

```diff
@@ -1618,39 +1618,116 @@ function cleanRepo() {
 }
 
 /**
- * This method is called when the sync button is pressed, and causes the fetch-modal
- * to appear on the screen.
+ * This method is called when the user clicks the "Remote" button on the navbar.
  */
-function requestLinkModal() {
-  $("#fetch-modal").modal();
+function setUpstreamModal() {
+  $('#set-upstream-modal').modal('show');
+  let repository;
+  Git.Repository.open(repoFullPath)
+      .then(function (repo) {
+        repository = repo;
+        Git.Remote.lookup(repository, 'upstream').then(function(remote) {
+          document.getElementById("display-upstream").innerText = remote.url();
+          document.getElementById("deleteUpstreamBtn").style.display = "block";
+        }, function(err) {
+          document.getElementById("display-upstream").innerText = 'No upstream repository currently configured.';
+          document.getElementById("deleteUpstreamBtn").style.display = "none";
+        });
+    });
 }
 
 /**
- * This method is called when a valid URL is given via the fetch-modal, and runs the
- * series of git commands which fetch and merge from an upstream repository.
+ * Clears the fields from the upstream repo modal.
  */
-function fetchFromOrigin() {
-  console.log("begin fetching");
-  let upstreamRepoPath = document.getElementById("origin-path").value;
-  if (upstreamRepoPath != null) {
+function clearUpstreamModalText() {
+  document.getElementById("upstream-path").value = "";
+}
+
+/**
+ * Delete the current upstream repo.
+ */
+function showUpstreamDelete() {
+  $('#delete-upstream-modal').modal('show');
+}
+
+/**
+ * Delete the current upstream repo.
+ */
+function deleteUpstream() {
```

Comment:
If i enter something into the upstream box:
![image](https://user-images.githubusercontent.com/15122789/59652739-e312b080-9143-11e9-90c3-2e9639e0ada9.png)

Then I git 'delete' then I either 'cancel' or 'confirm' then the status dialog is still filled out.

Reply:
The modal should now clear. What do you mean by deleting and setting the upstream in one command?

Reply:
If I enter a new path and hit confirm it should say "upstream already exists, do you want to replace with blah?" confirm/close ... and then it deletes and updates behind the scene?

Reply:
Sure, l'll create an issue and work on it in the next sprint.

Reply:
Issue #172